libraries/AP_HAL_ChibiOS/RCOutput.cpp

```diff
@@ -969,6 +978,54 @@ void RCOutput::dshot_send(pwm_group &group, bool blocking)
 #endif //#ifndef DISABLE_DSHOT
 }
 
+
+
+
+/*
+  fill in a DMA buffer for Neopixel LED
+ */
+void RCOutput::fill_DMA_buffer_neopixel(uint32_t *buffer, const uint8_t stride, uint32_t rgb, const uint16_t clockmul)
+{
+    // TODO: chane clockmul to a define or constant
+    const uint32_t DSHOT_MOTOR_BIT_0 = 7 * clockmul;
+    const uint32_t DSHOT_MOTOR_BIT_1 = 14 * clockmul;
+
+    for (uint16_t i=0; i < 24; i++) {
+        buffer[i * stride] = (rgb & 0x800000) ? DSHOT_MOTOR_BIT_1 : DSHOT_MOTOR_BIT_0;
+        rgb <<= 1;
+    }
+}
+
+/*
+  send a set of Neopixel packets for a channel group
+ */
+void RCOutput::neopixel_send(pwm_group &group)
+{
+#ifndef DISABLE_DSHOT
+    if (irq.waiter || !group.dma_handle->lock_nonblock()) {
+        // doing serial output, don't send Neopixel pulses
+        return;
+    }
+
+    memset((uint8_t *)group.dma_buffer, 0, neopixel_buffer_length);
+    
+    for (uint8_t i=0; i<4; i++) {
+        uint8_t chan = group.chan[i];
+        if (chan != CHAN_DISABLED) {
+            fill_DMA_buffer_neopixel(group.dma_buffer + i, 4, neopixel_rgb_data[i], group.bit_width_mul);
+        }
+    }
+
+    // start sending the pulses out
+    send_pulses_DMAR(group, neopixel_buffer_length);
+
+    group.last_dshot_send_us = AP_HAL::micros64();
```

Comment:
rename last_dshot_send_us to last_dmar_send_us

Reply:
fixed